## *********************************************************
## 
## File autogenerated for the sl_controller package 
## by the dynamic_reconfigure package.
## Please do not edit.
## 
## ********************************************************/

##**********************************************************
## Software License Agreement (BSD License)
##
##  Copyright (c) 2008, Willow Garage, Inc.
##  All rights reserved.
##
##  Redistribution and use in source and binary forms, with or without
##  modification, are permitted provided that the following conditions
##  are met:
##
##   * Redistributions of source code must retain the above copyright
##     notice, this list of conditions and the following disclaimer.
##   * Redistributions in binary form must reproduce the above
##     copyright notice, this list of conditions and the following
##     disclaimer in the documentation and/or other materials provided
##     with the distribution.
##   * Neither the name of the Willow Garage nor the names of its
##     contributors may be used to endorse or promote products derived
##     from this software without specific prior written permission.
##
##  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
##  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
##  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
##  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
##  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
##  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
##  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
##  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
##  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
##  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
##  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
##  POSSIBILITY OF SUCH DAMAGE.
##**********************************************************/

from dynamic_reconfigure.encoding import extract_params

inf = float('inf')

config_description = {'upper': 'DEFAULT', 'lower': 'groups', 'srcline': 233, 'name': 'Default', 'parent': 0, 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'cstate': 'true', 'parentname': 'Default', 'class': 'DEFAULT', 'field': 'default', 'state': True, 'parentclass': '', 'groups': [], 'parameters': [{'srcline': 259, 'description': 'What kind of supply power will be used', 'max': 2, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'Supply_type', 'edit_method': "{'enum_description': 'Supply type', 'enum': [{'srcline': 16, 'description': 'Supply: Wire', 'srcfile': '../cfg/reconfigure.cfg', 'cconsttype': 'const int', 'value': 0, 'ctype': 'int', 'type': 'int', 'name': 'Wire'}, {'srcline': 17, 'description': 'Supply: battery 2200 mAh', 'srcfile': '../cfg/reconfigure.cfg', 'cconsttype': 'const int', 'value': 1, 'ctype': 'int', 'type': 'int', 'name': 'Battery_2200_mAh'}, {'srcline': 18, 'description': 'Supply: battery 2700 mAh', 'srcfile': '../cfg/reconfigure.cfg', 'cconsttype': 'const int', 'value': 2, 'ctype': 'int', 'type': 'int', 'name': 'Battery_2700_mAh'}]}", 'default': 0, 'level': 2, 'min': 0, 'type': 'int'}, {'srcline': 259, 'description': 'Load hanging at the copter', 'max': 2000.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'Load_0_200', 'edit_method': '', 'default': 0.0, 'level': 2, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'frequency of the HLC', 'max': 50.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'Frequency', 'edit_method': '', 'default': 30.0, 'level': 2, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'set the initial parameter (from file controller_patameter.yaml)', 'max': True, 'cconsttype': 'const bool', 'ctype': 'bool', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'init_PID', 'edit_method': '', 'default': False, 'level': 1, 'min': False, 'type': 'bool'}, {'srcline': 259, 'description': 'load parameter from a file or record the courrent', 'max': True, 'cconsttype': 'const bool', 'ctype': 'bool', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'load_param', 'edit_method': '', 'default': False, 'level': 1, 'min': False, 'type': 'bool'}, {'srcline': 259, 'description': 'use the loaded or saved parameter', 'max': True, 'cconsttype': 'const bool', 'ctype': 'bool', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'use_param', 'edit_method': '', 'default': False, 'level': 1, 'min': False, 'type': 'bool'}, {'srcline': 259, 'description': 'K_P z PID controller', 'max': 300.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'thrust_K_P', 'edit_method': '', 'default': 100.0, 'level': 2, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'K_I z PID controller', 'max': 30.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'thrust_K_I', 'edit_method': '', 'default': 10.0, 'level': 2, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'K_D z PID controller', 'max': 150.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'thrust_K_D', 'edit_method': '', 'default': 50.0, 'level': 2, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'P_action max z PID controller', 'max': 100.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'thrust_max_P', 'edit_method': '', 'default': 30.0, 'level': 2, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'I_action max z PID controlle', 'max': 100.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'thrust_max_I', 'edit_method': '', 'default': 30.0, 'level': 2, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'D_action max z PID controlle', 'max': 100.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'thrust_max_D', 'edit_method': '', 'default': 40.0, 'level': 2, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'PID_action max z PID controlle', 'max': 500.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'thrust_max_PID', 'edit_method': '', 'default': 200.0, 'level': 2, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'K_P cross position controller', 'max': 2.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'cross_pos_K_P', 'edit_method': '', 'default': 0.6, 'level': 2, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'P_action max corss position controller', 'max': 3.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'cross_pos_max_P', 'edit_method': '', 'default': 0.8, 'level': 2, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'K_P cross rate controller', 'max': 1.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'cross_rate_K_P', 'edit_method': '', 'default': 0.2, 'level': 2, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'K_D cross rate controller', 'max': 0.1, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'cross_rate_K_D', 'edit_method': '', 'default': 0.0, 'level': 2, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'P_action max cross rate controller', 'max': 1.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'cross_rate_max_P', 'edit_method': '', 'default': 0.2, 'level': 2, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'D_action max cross rate controller', 'max': 1.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'cross_rate_max_D', 'edit_method': '', 'default': 0.2, 'level': 2, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'PD_action max cross rate controller', 'max': 1.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'cross_rate_max_PD', 'edit_method': '', 'default': 0.2, 'level': 2, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'K_P along position controller', 'max': 2.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'along_pos_K_P', 'edit_method': '', 'default': 0.6, 'level': 2, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'P_action max along position controller', 'max': 3.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'along_pos_max_P', 'edit_method': '', 'default': 0.8, 'level': 2, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'K_P along rate controller', 'max': 1.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'along_rate_K_P', 'edit_method': '', 'default': 0.2, 'level': 2, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'K_D along rate controller', 'max': 0.1, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'along_rate_K_D', 'edit_method': '', 'default': 0.0, 'level': 2, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'P_action max along rate controller', 'max': 1.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'along_rate_max_P', 'edit_method': '', 'default': 0.2, 'level': 2, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'D_action max along rate controller', 'max': 1.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'along_rate_max_D', 'edit_method': '', 'default': 0.2, 'level': 2, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'PD_action max along rate controller', 'max': 1.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'along_rate_max_PD', 'edit_method': '', 'default': 0.2, 'level': 2, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'Numbers of loops to wait untill feeding the angles', 'max': 20, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'N_Delay', 'edit_method': '', 'default': 3, 'level': 2, 'min': 0, 'type': 'int'}, {'srcline': 259, 'description': 'Gain on the load angles', 'max': 0.5, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'G_d', 'edit_method': '', 'default': 0.0, 'level': 2, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'length from the copter CoG to puppet CoG', 'max': 2.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'l_L', 'edit_method': '', 'default': 0.6, 'level': 2, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'Sign for delay cross track', 'max': 1.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'df_sign_ct', 'edit_method': '', 'default': 1.0, 'level': 2, 'min': -1.0, 'type': 'double'}, {'srcline': 259, 'description': 'Sign for delay along track', 'max': 1.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'df_sign_at', 'edit_method': '', 'default': 1.0, 'level': 2, 'min': -1.0, 'type': 'double'}, {'srcline': 259, 'description': 'turn input shaper on and off', 'max': 1, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'is_on', 'edit_method': '', 'default': 0, 'level': 2, 'min': 0, 'type': 'int'}, {'srcline': 259, 'description': 'Number of impulses', 'max': 5, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'is_N_impulses', 'edit_method': '', 'default': 2, 'level': 2, 'min': 1, 'type': 'int'}, {'srcline': 259, 'description': 'Frequency cross track', 'max': 10.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'is_omega_ct', 'edit_method': '', 'default': 0.5, 'level': 2, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'Frequency along track', 'max': 10.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'is_omega_at', 'edit_method': '', 'default': 0.5, 'level': 2, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'damping cross track', 'max': 1.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'is_zeta_ct', 'edit_method': '', 'default': 0.1, 'level': 2, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'damping along track', 'max': 1.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'is_zeta_at', 'edit_method': '', 'default': 0.1, 'level': 2, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'debug channel 1', 'max': 100.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'test1', 'edit_method': '', 'default': 0.0, 'level': 2, 'min': -100.0, 'type': 'double'}, {'srcline': 259, 'description': 'debug channel 2', 'max': 100.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'test2', 'edit_method': '', 'default': 0.0, 'level': 2, 'min': -100.0, 'type': 'double'}, {'srcline': 259, 'description': 'debug channel 3', 'max': 100.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'test3', 'edit_method': '', 'default': 0.0, 'level': 2, 'min': -100.0, 'type': 'double'}, {'srcline': 259, 'description': 'initialize the coordinate system where the copter in this moment is', 'max': True, 'cconsttype': 'const bool', 'ctype': 'bool', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'init_coordinate', 'edit_method': '', 'default': False, 'level': 1, 'min': False, 'type': 'bool'}], 'type': '', 'id': 0}

min = {}
max = {}
defaults = {}
level = {}
type = {}
all_level = 0

#def extract_params(config):
#    params = []
#    params.extend(config['parameters'])    
#    for group in config['groups']:
#        params.extend(extract_params(group))
#    return params

for param in extract_params(config_description):
    min[param['name']] = param['min']
    max[param['name']] = param['max']
    defaults[param['name']] = param['default']
    level[param['name']] = param['level']
    type[param['name']] = param['type']
    all_level = all_level | param['level']

reconfigure_INIT_FILTER = 1
reconfigure_MISC = 2
reconfigure_Wire = 0
reconfigure_Battery_2200_mAh = 1
reconfigure_Battery_2700_mAh = 2
